### DISCUSSION TITLE
Welcome to zkync-developers Discussions!

### DISCUSSION THREAD

**Discussion by bxpana** [zkSync-Community-Hub/zksync-developers#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.
* Use the Discussions as a resource to collaborate with peers and find answers from the community. 

## ❓ Questions
The Questions Section of this Github Discussion is the perfect space for you to ask questions and seek answers from our vibrant and knowledgeable community.

Here, you'll find several categories, each dedicated to a specific topic. Whether your question pertains to SDKs, Contract Verification, Deployments, or another subject, you'll find a relevant category where you can initiate a discussion.
- [Compilers](https://github.com/zkSync-Community-Hub/zkync-developers/discussions/categories/compilers)
- [SDKs](https://github.com/zkSync-Community-Hub/zkync-developers/discussions/categories/sdks)
- [Contract Verification](https://github.com/zkSync-Community-Hub/zkync-developers/discussions/categories/contract-verification)
- [Deployments](https://github.com/zkSync-Community-Hub/zkync-developers/discussions/categories/deployments)
- [General](https://github.com/zkSync-Community-Hub/zkync-developers/discussions/categories/general)

**Comment by 0xRektor** [comment]
An interesting experiment. I'll be watching

**Reply by nlalkmaar** [reply]
watch

**Reply by zkssui9** [reply]
3填VB

**Reply by smolNBA** [reply]
An interesting. This is the kind of discussion I'm most comfortable with

**Reply by yog2012** [reply]
watch

**Reply by Ali3447** [reply]
watch

**Reply by monymen** [reply]
watch

**Reply by gokopotter** [reply]
watch

**Reply by otooy** [reply]
how

**Reply by jmwhangjin** [reply]
Gonna Watch

**Reply by zssbecker** [reply]
watch!!

**Reply by deppouriaht** [reply]
watch

**Reply by mahdi00007** [reply]
wach

**Reply by derasu** [reply]
wach

**Reply by CryptoEmko** [reply]
Wach

**Reply by kamatu** [reply]
watch

**Reply by l3erryy99** [reply]
watch

**Reply by nftk3vin** [reply]
+1

**Reply by Irbidonles** [reply]
hay.... any link DC ZkSycn i want to join .......
link on guild and official web of zksync has expied......
please help me......

**Reply by zerosecxz** [reply]
+1

**Reply by Clemize** [reply]
nice

**Reply by ahmedmandour1** [reply]
good chance to learn I will watching

**Reply by kahkaha54** [reply]
bizde izleyelim :rotating_light:

**Reply by toto1975** [reply]
good luck!!!

**Reply by Ea1154** [reply]
nice

**Reply by deesja** [reply]
nice

**Reply by Islammohamedashraf** [reply]
Good luck 🤞

**Reply by RogerSterlings** [reply]
Follow

**Reply by nahomtz** [reply]
nice

**Reply by Draugin** [reply]
first time to use github. Thank to zksync. WAGMI

**Reply by P62PPP** [reply]
hallo everybody. Can somebody help? I can not connect to the discord of zksync, i try from few browsers, i am trying from zksync.io invitation
thanks in advance

**Reply by MrRedl** [reply]
nice nice!

**Reply by enerelt0418** [reply]
glad to join here mates

**Comment by lellycrypto-code** [comment]
https://github.com/matter-labs/zksync/blob/v0.7.1/contracts/interfaces/IZkSync.sol

What's the solution to this problem

**Reply by lellycrypto-code** [reply]
@bxpana @0xRektor

**Reply by VOVANCAO** [reply]
I'm connecting to confirm my wallet but I can't link it, even though I have enough points for gitcoin. Can you help with instructions?
![guild](https://github.com/zkSync-Community-Hub/zksync-developers/assets/116540121/35198546-b018-4111-8a40-5fd234c0e760)

**Reply by bxpana** [reply]
You'll want to reach out to Guild since it's their platform.

**Reply by jaguard2021** [reply]
@VOVANCAO  
![aasasas](https://github.com/zkSync-Community-Hub/zksync-developers/assets/88972434/0807532a-49c3-4815-9b31-b3d4f21fa498)

**Reply by Irbidonles** [reply]
hay.... any link DC ZkSycn i want to join .......
link on guild and official web of zksync has expied......
please help me......

**Reply by Draugin** [reply]
I couldnt get level 1 on guild. I hope its temporary

**Reply by doypro** [reply]
Guys I have a problem here, regarding Discord Server. I joined on Discord Server since Feb 2023, unfortunately, today I can't find the Server on my Discord, I have a tendency of opening zkSync server almost every single day, even if I go to read for others conservations, so dont believe that I did any mistake to be kicked out. My problem is, I can't join the Server any more, every link I try it expired!! What can I do to access the Server again?? 
Help me please.

Discord name:
Doypro.Official

**Comment by damien75** [comment]
Discussions on Github with message version tracking (meant to be), that's a good idea. Interested to see how this goes 👀

**Reply by Clemize** [reply]
Same sir

**Reply by Chizimminista** [reply]
hmmm ure just joining now

**Comment by verma-shu6ham** [comment]
Hello, I have some questions that I would appreciate if someone could help clarify:

1. Suppose I have an ERC20 smart contract currently on the Ethereum network, and I wish to migrate it to zkSync. Would this necessitate the use of a bridge? If so, does this imply that any token can be bridged? Specifically, if I deploy the ERC20 to the Goerli network for testing purposes, am I able to bridge it to zkSync myself, or is this something only the zkSync team can do?

2. Is the migration limited to ERC20 tokens, or can other types of smart contracts also be migrated? If the latter, what is the appropriate procedure? Additionally, during the migration, is it possible to move the storage of the existing Goerli contract to zkSync?

**Reply by dutterbutter** [reply]
Hello 👋 

If your token's already on Ethereum, you can bridge it with a script that uses the default bridge. You can find the deposit and withdraw scripts in the related section of our [documentation](https://era.zksync.io/docs/reference/concepts/bridging/bridging-asset.html) :book:.

By using the default bridge, if your token hasn't travelled to L2 before, our system automatically deploys a default ERC20 token contract on L2 for you, saving you the manual labor :toolbox:. However, be aware, specific functionalities or methods from your original L1 contract won't transfer over to L2 with this method. 

If you want to deploy your existing ERC20 token contract or any other contract to zkSync Era you can do so as well, please refer to our [Build](https://era.zksync.io/docs/dev/#getting-started) section in the docs to get started :muscle:.

Or, you could build your own bridge :bridge_at_night: and deploy the token contract on L2 yourself! [L1-L2 messaging](https://era.zksync.io/docs/reference/concepts/bridging/l1-l2-interop.html) can be used for this. Happy hacking!

**Reply by verma-shu6ham** [reply]
Hello @dutterbutter,

I would like to thank you for addressing my previous query. In your response, you provided specific information regarding ERC20 tokens. While I will be exploring that, I also have some questions about another custom smart contract example provided below.

```
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.0;

contract CustomContract {
    mapping(address => bool) private authorizedPersons;
    address private owner;

    constructor() {
        owner = msg.sender;
    }

    modifier onlyOwner() {
        require(msg.sender == owner, "Only owner.");
        _;
    }

    modifier onlyAuthorizedPerson() {
        require(authorizedPersons[msg.sender], "Only authorized persons have access.");
        _;
    }

    function getPersonIsAuthorized(address _address) external view returns (bool) {
        return authorizedPersons[_address];
    }

    function addPerson(address _address) external onlyOwner returns (bool) {
        require(!authorizedPersons[_address], "Already authorized.");
        authorizedPersons[_address] = true;
        return true;
    }

    function removePerson(address _address) external onlyOwner returns (bool) {
        require(authorizedPersons[_address], "Not authorized.");
        authorizedPersons[_address] = false;
        return true;
    }
}
```

Suppose I need to migrate this contract to zkSync while preserving its storage. How can I achieve this? Additionally, I would like to know if the old smart contract on the Ethereum network will still be functional after migration, or if only the zkSync smart contract will work.

**Reply by Bensmithz07** [reply]
Nice performance

**Reply by yog2012** [reply]
Nice

**Reply by CryptoEmko** [reply]
Very nice

**Reply by evmparser** [reply]
looks good!

**Reply by bob456456** [reply]
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.0;

contract CustomContract {
    mapping(address => bool) private authorizedPersons;
    address private owner;

    event PersonAdded(address indexed person);
    event PersonRemoved(address indexed person);

    constructor() {
        owner = msg.sender;
    }

    modifier onlyOwner() {
        require(msg.sender == owner, "Only owner.");
        _;
    }

    modifier onlyAuthorizedPerson() {
        require(authorizedPersons[msg.sender], "Only authorized persons have access.");
        _;
    }

    function getPersonIsAuthorized(address _address) external view returns (bool) {
        return authorizedPersons[_address];
    }

    function addPerson(address _address) external onlyOwner returns (bool) {
        require(!authorizedPersons[_address], "Already authorized.");
        authorizedPersons[_address] = true;
        emit PersonAdded(_address);
        return true;
    }

    function removePerson(address _address) external onlyOwner returns (bool) {
        require(authorizedPersons[_address], "Not authorized.");
        authorizedPersons[_address] = false;
        emit PersonRemoved(_address);
        return true;
    }
}

**Comment by starkqwa** [comment]
Interesting thanks for you responding

**Comment by vishnu5104** [comment]
Hi, 

I'm conducting some research on the best NFT Fetching/mapping API to obtain on-chain data. I'd appreciate your insights and recommendations. Could you please share your experiences and suggestions regarding which API works most effectively for this purpose?

**Reply by SooparApe** [reply]
Yes, I would like to know this too!

I usually use Alchemy or Moralis but niether support NFT mapping on ZKSync

**Comment by richiebinance** [comment]
Long life zksync 💕

**Comment by foodara90** [comment]
boom to zksync

**Comment by artem199004** [comment]
HELLO. I can’t join the discord nickname Maladoy#0639. Tell me how to solve this problem. Thank you

**Reply by bxpana** [reply]
Please email us at support@zksync.io and we can help you there.

**Reply by ashik0121** [reply]
Sir, I cannot join the discord server.. Plz help me

**Reply by Adrenalin1989** [reply]
Привіт. Чи вдалось вирішити проблему? Теж саме, не можу приєднатися до дискорду. Дякую

**Comment by ashik0121** [comment]
Sir, I cannot join the discord server.. Plz help me

**Comment by edmundedgar** [comment]
Hi, a while back I ran into this issue where if you use `transfer()` on zksync it fails because it hard-codes a gas assumption and that assumption doesn't hold on zksync. Then IIRC somebody at zksync invented a clever solution that would allow it to still work on zksync. Did that solution ever get deployed? If not do we know approximately when it might be?

(I don't have funds stuck or anything and it's easy to work around it, I'd just rather deploy the code with transfer() because it's audited and deployed that way on other chains.)

**Reply by bxpana** [reply]
Check out https://github.com/zkSync-Community-Hub/zksync-developers/discussions/229#discussioncomment-7927061 for the most recent update

**Comment by yog2012** [comment]
I am very glad to find the development community,I join

**Reply by 0xnth** [reply]
🫡🫡

**Comment by jaguard2021** [comment]
Where can I find information on how to verify a contract in ZKSync or useful guidelines for the verification process? Thank you for the assistance from all community members

**Reply by bxpana** [reply]
The best spot will be in our docs https://era.zksync.io/docs/dev/how-to/verify-contracts.html#verify-contracts-with-hardhat

**Reply by bxpana** [reply]
If you have any questions or issues please feel free to open up a discussion post 😁

**Reply by jaguard2021** [reply]
Thanks Sir

**Comment by yog2012** [comment]
hey,how can I submit the code?

**Comment by arnolwijaya77** [comment]
why total transactions in etherscan version shown less compared to the previous one?

**Comment by Sebuska29190** [comment]
Need DISCORD working link to join. The one on the website doesn't work

**Reply by bxpana** [reply]
It does, if you're having issues please email us at support@zksync.io

**Reply by hapizddcr** [reply]
I think u discord has banned. U can appeal here : https://appeal.zksync.dev/

**Comment by earnings4all** [comment]
Will there be another chance for minting LIBERTAS OMNIBUS as it was mentioned earlier in twitter?

**Reply by hapizddcr** [reply]
Maybe

**Comment by Adinurhuda** [comment]
Hello. Excuse me, I want to submit a request regarding unban my discord. The chronology is that I was hit by a both attack so it was detected as violating the zksync discord rules. I have participated in many zksync events such as transactions in zksync era and transactions in zksync lite. Please consider my discord unban request

username : liger1008
user ID: 1098561155389194240

thank you.

**Reply by hapizddcr** [reply]
Appeal Here https://appeal.zksync.dev/

**Reply by addcrypto1** [reply]
Hi, it's been a long time since I've been able to access the discod channel. Please can you help me?

**Reply by bxpana** [reply]
@addcrypto1 Please email us at support@zksync.io and we can help you with your Discord issues there. This is for Developer support

**Reply by Adinurhuda** [reply]
I finished filling in a week ago, when will I get a reply? until now I can't join the zksync discord?

**Comment by bryannguyen1812** [comment]
Hi, i got issue of transaction. need help pls

**Reply by petarTxFusion** [reply]
Hello, 
I suggest that you make new discussion with the problem you are having and post as much details as you can.

**Comment by bryannguyen1812** [comment]


**Reply by hapizddcr** [reply]
What?

**Comment by wlad010180** [comment]
![Знімок екрана (270)](https://github.com/zkSync-Community-Hub/zksync-developers/assets/134652636/a5fded2f-8b9a-49b8-9ad7-ff1ab8c3c01a)
![Знімок екрана (271)](https://github.com/zkSync-Community-Hub/zksync-developers/assets/134652636/a11b595c-c196-4ce4-9bc0-1807f7f5987c)

**Reply by wlad010180** [reply]
Hello , tell me why the Gitcoin passport is not verified

**Reply by hapizddcr** [reply]
Click x on gitcoin and 🔃 / recheck

**Reply by addcrypto1** [reply]
Hi, it's been a long time since I've been able to access the discod channel. Please can you help me?

**Reply by vladpr1703** [reply]
What's the problem? Describe it in detail.

**Reply by jaguard2021** [reply]
You have to press the small refresh button on the top right

**Comment by mnuww** [comment]
thank i wil join

**Comment by addcrypto1** [comment]
HI,
I haven't been able to access the zkSync server on discord for a long time.
Please can you help me?
When I follow the link to access the Discord server it says: "Failed to
accept invitation" (see screenshot1).

If I can't log in to the server I won't be able to finish GUIL (screenshot2)

Thanks for any help you can give me.

Anthony

My Discord account: addcrypto1#7472


Il giorno gio 15 feb 2024 alle ore 21:29 Vlad. ***@***.***>
ha scritto:

**Reply by hapizddcr** [reply]
Appeal here : https://appeal.zksync.dev/

**Comment by Xazrael** [comment]
Why i can't deploy that standart script what should i do to fix that gas issue ? I tried to increase gweis and gasses nothing changed.

![image](https://github.com/zkSync-Community-Hub/zksync-developers/assets/20703061/bdcc5a3c-e1e0-439e-b834-7d58b1465c40)


// SPDX-License-Identifier: GPL-3.0

pragma solidity >=0.8.2 <0.9.0;

/**
 * @title Storage
 * @dev Store & retrieve value in a variable
 * @custom:dev-run-script ./scripts/deploy_with_ethers.ts
 */
contract Storage {

    uint256 number;

    /**
     * @dev Store value in variable
     * @param num value to store
     */
    function store(uint256 num) public {
        number = num;
    }

    /**
     * @dev Return value 
     * @return value of 'number'
     */
    function retrieve() public view returns (uint256){
        return number;
    }
}

**Comment by earnings4all** [comment]
Good 👍

**Comment by mankzan** [comment]
What docker volume driver is used in zkSync, can anybody help me with this ?

**Comment by sunarus** [comment]
Nice

**Comment by Arlo118** [comment]
keep refreshing and wait and be patient!

**Comment by FrondEnt** [comment]
I think idea with gitcoin passports = very smart to be eligible for Success programm

**Reply by bob456456** [reply]
Agree ! Great idea

**Reply by Ea1154** [reply]
gitcoin=success

**Reply by sunarus** [reply]
Must be using gitcoin passport or guild lvl 1 minimum for sybil filter every wallet

**Reply by zerosecxz** [reply]
Gitcoin is best option but minimum amount $ wallet holder also good for filter sybil...

**Comment by AgataGolik** [comment]
Hello everyone,
please explain to me at what stage we are now and are there any further deadlines for the various stages ? Lite, Era, Boojum or zkPorter ? I know that Lite and Era is already , while the other two ?

**Comment by Chizimminista** [comment]
> ## 👋 Welcome!
> We’re using Discussions as a place to connect with other members of our community. We hope that you:
> 
> * Ask questions you’re wondering about.
> * Share ideas.
> * Engage with other community members.
> * Welcome others and are open-minded. Remember that this is a community we
>   build together 💪.
> * Use the Discussions as a resource to collaborate with peers and find answers from the community.
> 
> ## ❓ Questions
> The Questions Section of this Github Discussion is the perfect space for you to ask questions and seek answers from our vibrant and knowledgeable community.
> 
> Here, you'll find several categories, each dedicated to a specific topic. Whether your question pertains to SDKs, Contract Verification, Deployments, or another subject, you'll find a relevant category where you can initiate a discussion.
> 
> * [Compilers](https://github.com/zkSync-Community-Hub/zkync-developers/discussions/categories/compilers)
> * [SDKs](https://github.com/zkSync-Community-Hub/zkync-developers/discussions/categories/sdks)
> * [Contract Verification](https://github.com/zkSync-Community-Hub/zkync-developers/discussions/categories/contract-verification)
> * [Deployments](https://github.com/zkSync-Community-Hub/zkync-developers/discussions/categories/deployments)
> * [General](https://github.com/zkSync-Community-Hub/zkync-developers/discussions/categories/general)

to be honest after going through zksync ecosystem and trying out it features, im still yet to understand the purpose of zklite.
pls enlighten me in any way

**Comment by akamanfc** [comment]
cool news

**Comment by franksofa** [comment]
This initiative to utilize GitHub Discussions as a platform for community interaction and knowledge sharing is commendable. It provides a structured and accessible space for members to ask questions, share ideas, and collaborate effectively. By categorizing discussions into relevant topics such as Compilers, SDKs, Contract Verification, Deployments, and General, it facilitates efficient navigation and ensures that discussions remain focused and productive. Overall, leveraging GitHub Discussions in this manner fosters a vibrant and supportive community environment, encouraging active participation and mutual learning among its members.

**Comment by stennettmeisha** [comment]
Welcome to the community! Feel free to ask questions, share ideas, and engage with fellow members here in Discussions. Let's collaborate, learn, and grow together!

**Comment by yas3i3** [comment]
nice job

**Comment by sundaramjacob** [comment]
Welcome to our vibrant community hub on Github Discussions! 

 Ask questions, share ideas, and engage with fellow members.
 Find answers, collaborate on projects, and explore new topics.
 Let's build and learn together as a community!

**Reply by Chizimminista** [reply]
are you a dev?

**Comment by kaiw0866** [comment]
I have a problem. I have submitted two transactions to the ETH main network respectively, and the first transaction cost is relatively low, resulting in withdrawal failure. Can I combine the two transactions? That way you don't have to pay as much GAS

**Reply by bxpana** [reply]
If you have an issue you can either open up a discussion or email us at support@zksync.io

**Comment by l3erryy99** [comment]
So nice roll-up mate!

**Comment by victorma17** [comment]
Hey there! We are a small startup trying to use a paymaster and We would like to know if is possible sponsorize a transaction using a paymaster from an Account that we don't have the Private Key, I mean, We don't if it is possible authorise one transaction through third party apps like Wallet Connect. Thanks in advance!

**Comment by sabarteros** [comment]
wow amazing.

**Comment by gerote** [comment]
can't wait for boom zk.

**Comment by dematre** [comment]
i joined

**Comment by cowokjantan** [comment]
Will the zksync era be able to use gas other than ETH like zksync lite???

**Comment by amir7970** [comment]
Hello sir. I have accidentally sent my zktokens to zksync contract address.
here is the transaction hash: 0x53752d5d23b26a35b4e85ea85b47a595a5508eebde150ef5b0806cb6fa846649
Could some one help me to get the tokens back?

**Comment by malingzhao** [comment]
2024-09-04T06:31:15.463957Z  INFO NamedFuture{name=node_storage_initializer}: zksync_node_framework::implementations::layers::node_storage_init: Node storage initialization task completed
Error: One or more tasks failed: [TaskFailed(TaskId("oneshot_runner"), Oneshot task node_storage_initializer_precondition failed

Caused by:
    0: Failed acquiring DB connection [requested by `en` at core/node/node_storage_init/src/external_node/snapshot_recovery.rs:75]: pool timed out while waiting for an open connection
    
    how to  reslove it  deploy node

**Comment by sarthak608** [comment]
Can someone here explain and share any resources or content that would help me understand ZKSync Prover such a boojum, belman, and other cryptography pairing library ? I am currently working on gaining a deeper understanding of ZK Provers and have come across ZKSync Prover. Any guidance or helpful materials would be greatly appreciated!

**Reply by dutterbutter** [reply]
Hey @sarthak608 👋  Check out the documentation below 👇 

https://matter-labs.github.io/zksync-era/prover/latest/

**Reply by sarthak608** [reply]
Hey @dutterbutter Thank you so much for providing a document having a deeper drive on ZKSync Prover.

**Reply by sarthak608** [reply]
Hey @dutterbutter 
Pls correct me if I am wrong, Need to confirm below high level understanding on ZKsync 

In zkSync, a Merkle tree manages the state of the network by organizing data into a structure of cryptographic hashes, making it efficient for verifications and representing the state of zkSync as a single hash. Zero-Knowledge Proofs are used to ensure the correct computation of state transitions from one Merkle root to the next (one state to another), providing secure and scalable off-chain processing.

Does this correct, merkle and ZKP use case in Zksync for network state management and verifying off-chain state change
